src/cloud.service.ts

```diff
@@ -722,12 +719,19 @@ export class CloudService {
     return this.get(url).pipe(map((p: Project) => new Project(p)));
   }
 
-  public fetchProjectWithAssociatedData(projectId: string): Observable<Project> {
+  public fetchProjectWithAssociatedData(projectId: string, apiFilter?: ApiFilter): Observable<Project> {
     return this.getProject(projectId).pipe(
-      mergeMap((p: Project) => this.fetchDevicesAndVariablesForProject(p).pipe(
+      mergeMap((p: Project) => {
+        let argsFilter;
```

Comment:
You should not need to do this. Simply pass apiFilter directly into the function, as apiFilter should either be undefined or it should have the filter. If undefined, it should be the same as not passing an argument.